scwidgets/converters.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Matthew Andrew
+
+import scipp as sc
+import pathlib
+
+scipp_object_mutable = (sc.DataArray, sc.Dataset, sc.Variable,
+                        sc.DataArrayView, sc.DatasetView, sc.VariableView)
+
+scipp_object = scipp_object_mutable + (
+    sc.DataArrayConstView, sc.DatasetConstView, sc.VariableConstView)
+
+
+def filepath_converter(filepath, data_directory=pathlib.Path.cwd()):
+    """
+    Checks the validity of the filepath provided,
+    returns it is valid. Otherwise tries to form a valid
+    filepath assuming a filename has been provided instead.
+
+    Throws if file not found.
+    """
+    path = pathlib.Path(filepath)
+
+    if path.is_file():
+        return filepath
+
+    path = data_directory / path
+
+    if path.is_file():
+        return str(path)
+
+    raise ValueError(f'Filepath {filepath} was not found.')
+
+
+def scope_converter(object_name, scope):
```

Comment:
What is this useful for?

Reply:
The inputs typed in the text-inputs are always strings. However a common case is that you are actually referring to an object in the current scope by name. This converter therefore returns the object of the given name in the given scope.

Reply:
Can we default to current or global scope or does that not make sense?

Reply:
Yes, getting access to the correct scope can be slightly non-trivial but we do do it for the module repr for example https://github.com/scipp/scipp/blob/master/python/src/scipp/object_list.py 

I'll have a look to see if there is a better way in this instance but if not the above does work.

Reply:
In this case just the global scope may work as you say.